### DISCUSSION TITLE
c内存没有释放

### DISCUSSION THREAD

**Discussion by walking-with-linght** [cloudwu/skynet#1671] [General]
目前是拿的别人现成的项目二次开发。
skynet用的最新版本
### 现象是：
玩家退出后，agent服务使用exit() （也用过kill）后，cmem任然有对应服务内存无法释放。
必现问题。
正在线上使用，同时目前最多四千多人，降到一千人后内存居高不下。
但是内存可以重复使用，比如：最高4000人，消耗内存50%，人数到1000人时，消耗内存还是50%，再次到高峰4000人时，内存还是50%。
技术现在只会lua，c一窍不通，懵逼很久，不敢提issues。ε=(´ο｀*)))

**Comment by sniper00** [comment]
只是标记这块内存还给系统，系统需要的时候再取

**Comment by hanxi** [comment]
http://cwqqq.com/2022/07/03/skynet_jemalloc_high_memory_use

**Comment by PowerGit** [comment]
这个是lua5.3的GC策略，可以手动调用collectgarbage("step", xxx)步进GC，加速内存释放

**Reply by PowerGit** [reply]
看错了，忽略，C内存要看jemalloc策略
